src/test/java/com/fasterxml/uuid/JugNamedTest.java

```diff
@@ -0,0 +1,178 @@
+        assertEquals('\n', actualUuid.charAt(actualUuid.length() - 1));
+
+        assertEquals(UUID.class,
+                UUID.fromString(actualUuid.substring(0, actualUuid.length() - 1)).getClass());
+    }
+
+    @Test
+    public void run_givenCount3_shouldProduceUUID() {
+        // given
+
+        // when
+        List<String> arguments = useCase.getArgs();
+        arguments.add(0, "-c");
+        arguments.add(1, "3");
+        jug_underTest.run(arguments.toArray((String[]) Array.newInstance(String.class, 0)));
+
+        // then - if it is a UUID then we should be able to parse it back out
+        String[] actualUuids = outContent.toString().split("\n");
+        for(String actualUuid: actualUuids) {
+            assertEquals(UUID.class,
+                    UUID.fromString(actualUuid).getClass());
+        }
+    }
+
+    @Test
+    public void run_givenPerformance_shouldProducePerformanceInfo() {
+        // given
+
+        // when
+        List<String> arguments = useCase.getArgs();
+        arguments.add(0, "-p");
+        jug_underTest.run(arguments.toArray((String[]) Array.newInstance(String.class, 0)));
+
+        // then
+        String actualOutput = outContent.toString();
+
+        assertThat(actualOutput, containsString("Performance: took"));
+    }
+    @Test
+    public void run_givenHelp_shouldProduceHelpInfo() {
+        // given
+
+        // when
+        List<String> arguments = useCase.getArgs();
+        arguments.add(0, "-h");
+        jug_underTest.run(arguments.toArray((String[]) Array.newInstance(String.class, 0)));
+
+        // then
+        String actualOutput = errContent.toString();
+
+        assertThat(actualOutput, containsString("Usage: java"));
+    }
+
+    @Test
+    public void run_givenVerbose_shouldProduceExtraInfo() {
+        // given
+
+        // when
+        List<String> arguments = useCase.getArgs();
+        arguments.add(0, "-v");
+        jug_underTest.run(arguments.toArray((String[]) Array.newInstance(String.class, 0)));
+
+        // then
+        String actualOutput = outContent.toString();
+
+        assertThat(actualOutput, containsString("Done."));
+    }
+
+    @Test
+    public void run_givenVerboseAndPerformance_shouldProduceExtraInfo() {
+        // given
+
+        // when
+        List<String> arguments = useCase.getArgs();
+        arguments.add(0, "-v");
+        arguments.add(1, "-p");
+        jug_underTest.run(arguments.toArray((String[]) Array.newInstance(String.class, 0)));
+
+        // then
+        String actualOutput = outContent.toString();
+
+        assertThat(actualOutput, containsString("Done."));
+        assertThat(actualOutput, containsString("Performance: took"));
+    }
+
+    @Parameterized.Parameters(name = "{index} -> {0}")
+    public static List<UseCase> useCases() {
+        return Arrays.asList(
+                new UseCase("n", "-n", "world", "-s", "url"),
+                new UseCase("n", "-n", "world", "-s", "dns")
```

Comment:
Here is some of the parameterized tests that I have applied, it looks kinda like this:

![image](https://github.com/cowtowncoder/java-uuid-generator/assets/8869415/3049334c-fcf1-4d53-b9c0-ffc5add46cf9)